runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/GreedyPipelineFuser.java

```diff
@@ -69,6 +69,14 @@ private GreedyPipelineFuser(Pipeline p) {
     fusePipeline(groupSiblings(rootConsumers));
   }
 
+  /**
+   * Fuses a {@link Pipeline} into a collection of {@link ExecutableStage}s.
+   *
+   * <p>This fuser expects each PTransform to have exactly one input. This means that pipelines must
```

Comment:
Not true; `Flatten` is permitted, as is `ParDo` with side inputs.

However, it does expect all transforms with no inputs to not have an associated environment, which is what forces this requirement.

Reply:
Fixed.